Formula/w/whisper-cpp.rb

```diff
@@ -22,11 +22,27 @@ class WhisperCpp < Formula
     sha256 cellar: :any_skip_relocation, x86_64_linux:   "f7187c90ab975f4558c1762a608e09bd16f897690b55308aa27bb6d21e86c2e2"
   end
 
-  def install
-    system "make"
-    bin.install "main" => "whisper-cpp"
+  depends_on "cmake" => :build
 
-    pkgshare.install ["samples/jfk.wav", "models/for-tests-ggml-tiny.bin", "ggml-metal.metal"]
+  def install
+    args = %W[
+      -DBUILD_SHARED_LIBS=OFF
+      -DGGML_METAL=#{(OS.mac? && !Hardware::CPU.intel?) ? "ON" : "OFF"}
+      -DGGML_METAL_EMBED_LIBRARY=#{OS.mac? ? "ON" : "OFF"}
+      -DGGML_NATIVE=#{build.bottle? ? "OFF" : "ON"}
+      -DWHISPER_BUILD_EXAMPLES=ON
+      -DWHISPER_BUILD_TESTS=OFF
+      -DWHISPER_BUILD_SERVER=OFF
+    ]
+    args << "-DLLAMA_METAL_MACOSX_VERSION_MIN=#{MacOS.version}" if OS.mac?
+
+    system "cmake", "-B", "build", *args, *std_cmake_args
+    # the "main" target is our whisper-cpp binary
+    system "cmake", "--build", "build", "--target", "main"
+    bin.install "build/bin/main" => "whisper-cpp"
```

Comment:
Is there no `install` target yet? `llama.cpp` seems to have one already these days.

Reply:
There is an install target but at the moment it only installs dylibs (had `BUILD_SHARED_LIBS` been left on) but not the `main` binary. I've submitted this as feedback to the devs and they are already updating their build system, so I expect this to improve in future releases.

Reply:
Then, we can just ship it as-is first and update it later.